Billing.API/Controllers/InvoiceController.cs

```diff
@@ -10,6 +12,8 @@ namespace Billing.API.Controllers
     [ApiController]
     public class InvoiceController : ControllerBase
     {
+        private static readonly string _invoiceFilenameRegex = @"^invoice_(\d){4}-(\d){2}-(\d){2}_(\d+)\.pdf$";
```

Comment:
I think that it will be cleaner if you do not create groups for information that you do not need to parse:

```suggestion
        private static readonly string _invoiceFilenameRegex = @"^invoice_\d{4}-\d{2}-\d{2}_(\d+)\.pdf$";
```

It will generate a change in the parsing line:

```
- var fileId = match.Groups[4].Value.ToInt32();
+ var fileId = match.Groups[1].Value.ToInt32();
```